### PR TITLE
DOCSP-29449 add telemetry information

### PR DESCRIPTION
**Pull request by davemungo**
[command line](https://docs-mongodbcom-staging.corp.mongodb.com/cluster-sync/docsworker-xlarge/DOCSP-29449-add-telemetry/reference/mongosync/#std-option-mongosync.--disableTelemetry)
[config file](https://docs-mongodbcom-staging.corp.mongodb.com/cluster-sync/docsworker-xlarge/DOCSP-29449-add-telemetry/reference/configuration/#mongodb-setting-disableTelemetry)
[telemetry](https://docs-mongodbcom-staging.corp.mongodb.com/cluster-sync/docsworker-xlarge/DOCSP-29449-add-telemetry/reference/telemetry/#std-label-c2c-telemetry)

[JIRA](https://jira.mongodb.org/browse/DOCSP-29499) [REP] Add Telemetry to mongosync

( Replaces and expands [DOCSP-28563](https://jira.mongodb.org/browse/DOCSP-28563), [PR 122](https://github.com/mongodb/docs-cluster-to-cluster-sync/pull/122/) )

[BUILD](https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=647e1cda67a273b816871b7e)

### CODE REVIEW

File: source/includes/opts/telemetry-statement.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+``mongosync`` collects anonymous, aggregated usage data to improve
+MongoDB products. ``mongosync`` collects this telemetry information by
+default, but you can also disable telemetry collection when you run
+``mongosync``.
```
[the remark this answers is not in the document]
Comment (davemungo) [reply source/includes/opts/telemetry-statement.rst]: Shortened a bit. I left the first part as a definition of what "telemetry" means in MDB context.

File: source/reference/configuration.txt [hunk 2]
```diff
@@ -68,6 +68,20 @@ Options
...
    see the :option:`--cluster1` option.
 
+.. setting:: disableTelemetry
+
+   *Type*: boolean
+
+   .. versionadded:: 1.4.0
+
+   .. include:: /includes/opts/disableTelemetry
+
```
[the remark this answers is not in the document]
Comment (davemungo) [reply source/reference/configuration.txt:78]: This was existing text - updated throughout the page

**Review by davemungo — COMMENTED** [review]
@sarah-olson-mongodb thanks for your feedback. Sending to external review next

**Comment by davemungo** [comment]
@ajayvijayakumar123 this ready for review when you have a moment.

File: source/includes/opts/telemetry-statement.rst [hunk 3]
```diff
@@ -0,0 +1,3 @@
+By default, ``mongosync`` collects anonymous, aggregated usage data to
```
Comment (ajayvijayakumar123) [inline source/includes/opts/telemetry-statement.rst:1]: I think we should explicitly mention that telemetry data would be sent to [Segment](https://segment.com/)

File: source/reference/telemetry.txt [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+- State information like when ``mongosync`` starts, starts committing,
+  or passes through intermediate states
+- Information about the source and destination clusters and the hardware
+  that hosts the ``mongosync`` instance
+- Initialization arguments
+- The number of source and destination writes
+- The estimated number of bytes ``mongosync`` copies
+- The estimated number of events ``mongosync`` applies
+
+``mongosync`` does not track:
```
Comment (ajayvijayakumar123) [inline source/reference/telemetry.txt:32]: Could we also add that mongosync does not track mongodb connection strings?

**Review by ajayvijayakumar123 — CHANGES_REQUESTED** [review]
Looks great! Left a few suggestions

File: source/includes/opts/telemetry-statement.rst (hunk 3 above)
Comment (ajayvijayakumar123) [reply source/includes/opts/telemetry-statement.rst:1]: gotcha that makes sense. I'm cool with leaving it out then

File: source/reference/telemetry.txt (hunk 4 above)
Comment (davemungo) [reply source/reference/telemetry.txt:32]: Done

**Review by davemungo — COMMENTED** [review]
@ajayvijayakumar123 Thanks for your feed back, this is ready for another look.

**Review by ajayvijayakumar123 — APPROVED** [review]
LGTM!

File: source/reference/configuration.txt [hunk 5]
```diff
@@ -56,34 +56,48 @@ Options
...
+   To set the ``disableTelemetry`` setting from the command line,
+   see the :option:`--disableTelemetry` option.
+
+   For more information see, :ref:`User Data Collection
```
Comment (jason-price-mongodb) [inline source/reference/configuration.txt:82]: Misplaced comma.

File: source/reference/mongosync.txt [hunk 6]
```diff
@@ -63,6 +63,18 @@ Global Options
...
+.. option:: --disableTelemetry
+
+   .. versionadded:: 1.4.0
+
+   .. include:: /includes/opts/disableTelemetry
+
+   To set the ``--disableTelemetry`` option from a configuration file,
+   see the :setting:`disableTelemetry` setting.
+
+   For more information see, :ref:`User Data Collection
```
Comment (jason-price-mongodb) [inline source/reference/mongosync.txt:75]: Misplaced comma.

File: source/reference/telemetry.txt [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+
+.. include:: /includes/opts/telemetry-statement.rst
+
+Data Collection
+---------------
+
+The data ``mongosync`` tracks includes:
+
+- State information like when ``mongosync`` starts, starts committing,
```
Comment (jason-price-mongodb) [inline source/reference/telemetry.txt:23]: Suggestion: Rewrite to not use "like when".

File: source/reference/telemetry.txt [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+   :depth: 2
+   :class: singlecol
+
+
+.. include:: /includes/opts/telemetry-statement.rst
+
+Data Collection
+---------------
+
+The data ``mongosync`` tracks includes:
```
Comment (jason-price-mongodb) [inline source/reference/telemetry.txt:21]: Suggestion: Rewrite to not use "tracks". Tracks has multiple meanings.

**Review by jason-price-mongodb — CHANGES_REQUESTED** [review]
@davemungo some issues and suggestions. Cheers.

**Review by jason-price-mongodb — APPROVED** [review]

**Pull request merged**